### ISSUE TITLE
QtGraphicalEffects "not installed" on Fedora 40.

### ISSUE THREAD

**Issue by theshatterstone** [MarianArlt/sddm-sugar-dark#26]
Hello.

I have SDDM preinstalled, on Fedora 40 Sway.

Output of ` qmake --version `:

``` 
QMake version 3.1
Using Qt version 6.7.1 in /usr/lib64 
```

I have the ` qt5-qtquickcontrols2 ` and ` qt5-qtgraphicaleffects-5.15.14-1` packages installed. 

When running the default version of the theme, I'm getting the errors that ```qtgraphicaleffects``` is not installed. After doing a further look into it, I tried running: 

```
QML2_IMPORT_PATH=/usr/lib64/qt5/qml QT_PLUGIN_PATH=/usr/lib64/qt5/plugins sddm-greeter --test-mode --theme sugar-dark-orig
```

and got the following errors:

``` 
Loading file:sugar-dark-orig/Main.qml...
file:///home/shatterstone/sugar-dark-orig/Main.qml:82:9: Type LoginForm unavailable 
             LoginForm { 
             ^
file:///home/shatterstone/sugar-dark-orig/Components/LoginForm.qml:41:5: Type Input unavailable 
         Input { 
         ^
file:///home/shatterstone/sugar-dark-orig/Components/Input.qml:23:1: The plugin '/usr/lib64/qt5/qml/QtGraphicalEffects/libqtgraphicaleffectsplugin.so' uses incompatible Qt library. (5.15.0) [release] 
     import QtGraphicalEffects 1.0 
     ^
file:///home/shatterstone/sugar-dark-orig/Main.qml:82:9: Type LoginForm unavailable 
             LoginForm { 
             ^
file:///home/shatterstone/sugar-dark-orig/Components/LoginForm.qml:41:5: Type Input unavailable 
         Input { 
         ^
file:///home/shatterstone/sugar-dark-orig/Components/Input.qml:23:1: The plugin '/usr/lib64/qt5/qml/QtGraphicalEffects/libqtgraphicaleffectsplugin.so' uses incompatible Qt library. (5.15.0) [release] 
     import QtGraphicalEffects 1.0 
     ^
Fallback to embedded theme 
```

I should mention I found a "solution" where I replaced every import of ` QtGraphicalEffects 1.0 ` in the theme with `import Qt5Compat.GraphicalEffects 1.0` which seems to mostly work (except that the user selector icon is broken; fixed it by setting ` visible: false ` in the "selectUser" ComboBox in Input.qml) so it mostly kinda works after some modifications, but I'm not sure why these are even necessary in the first place. Any help would be appreciated. If you need any more info, please let me know. Thanks.

Edit(s): Fixing formatting of code blocks

**Comment by udqy** [comment]
Hey! Facing the exact same issue rn. How did you  solve this?

**Comment by theshatterstone** [comment]
@udqy I basically rewrote the themes and removed Qtgraphicaleffects as a dependency, also removing their modules and some broken things in the process.It's very flaky, of course, and I'm against using software from random git repos, and I can't afford to provide support for this cut down version so I'll only link it if yoi explicitly ask for it.

Alternatively, follow what I did by using `sddm-greeter` in the terminal and reading the error messages to point you to which files have a `require` for Qtgraphicaleffects, then remove the require, run again, find and remove the code blocks related to it

**Comment by udqy** [comment]
Yeah i tried doing the same thing but couldnt make it work. Ended up switching themes. Thanks

**Comment by theshatterstone** [comment]
@udqy Which theme works for you? I couldn't find any that worked for me, as they all either used Qtgraphicaleffects or KDE Frameworks

**Comment by udqy** [comment]
[sddm-astronaut-theme](https://github.com/Keyitdev/sddm-astronaut-theme) works

**Comment by theshatterstone** [comment]
@udqy Thank you so much! I will not close this issue, however, and will let @MarianArlt close it if they so choose.

**Comment by ValeWalker27** [comment]
**My PC**:
- OS: Nobara Linux 42 (KDE Plasma Desktop Edition) x86_64
- Kernel: Linux 6.14.8-200.nobara.fc42.x86_64
- DE: KDE Plasma 6.3.5
- WM: KWin (Wayland)

Qt6 seems to be the default installed version in my PC running Nobara Linux 42 (KDE) and I got this theme working based on a couple of AI (Brave Leo AI) suggestions. 

For example, `import QtGraphicalEffects` should be replaced with `import Qt5Compat.GraphicalEffects` to work with Qt6. Additionally, themes should declare their Qt version compatibility by adding `QtVersion=6` to the `metadata.desktop` file.

Courtesy:
https://develop.kde.org/docs/plasma/theme/theme-porting-to-plasma6/
https://www.phoronix.com/news/SDDM-0.21-Display-Manager

Disclaimer: I am a novice and not sure if this change would break something else though.
